internal/test/e2e/run.go

```diff
@@ -592,3 +596,10 @@ func logTinkerbellTestHardwareInfo(conf *instanceRunConf, action string) {
 	}
 	conf.Logger.V(1).Info(action+" hardware for TestRunner", "hardwarePool", strings.Join(hardwareInfo, ", "))
 }
+
+func shuffleHardwareInventory(invCatalogue *hardwareCatalogue) {
```

Comment:
Why not make this a method in `hardwareCatalogue`? It's manipulating the internal extructure, it seems like a good idea to abstract that in a method instead of exposing it like this.

I fact, don't you need to use the mutex? If I'm not mistaken the `hardwareCatalogue` is shared between runner threads and all of them are going to try to call this method concurrently.

Reply:
Good catch! I should have seen this.